File: 240719/최소 점프 횟수/min-num-of-jumps.js
const fs = require("fs")
const input = fs.readFileSync(0).toString().trim().split("\n")
const n = Number(input[0])
const arr = input[1].split(" ").map(Number)

let stack =[[0, 0]]
let ans = Number.MAX_SAFE_INTEGER
let flag = false

while (stack.length > 0){
    let [idx, num] = stack.pop()
    // 점프하기 전에 먼저 도달했는지 검사를 해야 했는데 검사를 나중에 해서 도착지점인 0인곳에선 검사를 안해서 틀렸음
    if (idx === n - 1){
        ans = Math.min(ans, num)
        flag = true
    } else if (idx < n - 1){
        {
            for (let i = 1; i <= arr[idx]; i++){
                stack.push([idx + i, num + 1])
            } 
        }
    }
}

if (flag){
    console.log(ans)
} else {
    console.log(-1)
}

File: 240722/최소 점프 횟수/min-num-of-jumps.py
n = int(input())
arr = list(map(int, input().split()))

stack = [[0, 0]]
ans = float('inf')

while stack:
    [idx, num] = stack.pop()
    if idx >= n - 1:
        ans = min(ans, num)
    else:
        for i in range(1, arr[idx] + 1):
            stack.append([idx + i, num + 1])

if ans == float('inf'):
    print(-1)
else:
    print(ans)